packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -327,8 +347,18 @@ const DataGrid = forwardRef((props, ref) => {
     if (!isSection(toCell.row)) {
     // Obtain coordinate rectangles for grid container, column header, and new cell selection
       const gridContainerRect = tableContainerRef.current.getBoundingClientRect();
-      const columnHeaderRect = grid.current.rows[0].cells[toCell.col].getBoundingClientRect();
+      const columnHeaderRect = grid.current.rows[0].getBoundingClientRect();
+
+      // Set new cell column index value to be within Column Header and Column Header actions bounding indexes 
+      if(toCell.row === 0 && hasVisibleColumnHeaders && toCell.col > displayedColumns.length) {
+        toCell.col = displayedColumns.length - 1;
+      }
+      if(toCell.row === 1 && hasColumnHeaderActions && toCell.col > displayedColumns.length ) {
+        toCell.col = displayedColumns.length - 1;
+      }
+
```

Comment:
We have logic to handle the HOME and END scenarios in the handleKeyDown method.  The necessary correction should be there and not in handleMoveCellFocus.

Reply:
Moved this to handleKeyDown